server/src/main/java/io/crate/metadata/sys/TableHealthService.java

```diff
@@ -165,7 +167,23 @@ private TableHealth tableHealthFromEntry(Map.Entry<TablePartitionIdent, ShardsIn
         } catch (RelationUnknown e) {
             return null;
         }
-        return calculateHealth(ident, shardsInfo, tableInfo.numberOfShards());
+        int shardsCount = tableInfo.numberOfShards();
+        if (ident.partitionIdent != null) {
+            var partitionIndexName = IndexParts.toIndexName(
+                new RelationName(ident.tableSchema, ident.tableName),
+                ident.partitionIdent
+            );
+            var indexMetaData = clusterService.state().getMetadata().index(partitionIndexName);
+            assert indexMetaData != null : "index meta data for partition " + partitionIndexName + " not found";
```

Comment:
I think it's legitimate for the indexMetaData to be absent if the partition is deleted concurrently. We should probably return `null` if `indexMetaData` is null - similar to how we handle the `RelationUnknown` case.